lib/mito/utilities/SegmentedVector.h

```diff
@@ -54,9 +54,23 @@ namespace mito::utilities {
             }
         }
 
+        // const components accessor (random access, may return an invalid resource)
+        inline auto operator[](int i) -> resource_type &
+        {
+            // delegate the answer to the allocator
+            return _resources[i];
```

Comment:
This is risky. Might be good to do some boundary checks.

Reply:
I would leave the `operator[]` as a true wrapper to `operator[]` and perhaps implement `at` with bounds checking, for consistency with the STL vector.